integ-tests/src/main/java/com/google/sql/SqlHelper.java

```diff
@@ -83,4 +87,48 @@ public static void connectAndExecuteQueryAsUser(String username, String password
       throw e;
     }
   }
+
+  /**
+   * Returns count of rows for provided query.
+   *
+   * @param connection DB connection parameter
+   * @param query Select query with count entries names as RowsNR
+   */
+  public static int getEntriesCountForQuery(Connection connection, String query)
+      throws SQLException {
+    int count = 0;
+    try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+      ResultSet rs = preparedStatement.executeQuery();
+      rs.next();
+      count = getIntNotNull(rs, "RowsNR");
+    } catch (SQLException e) {
+      LOGGER.error(
+          format("Cannot execute query: %n%s%n", query));
+      throw e;
+    }
+    return count;
+  }
+
+  /**
+   * @param connection DB connection parameter
+   * @param query Select query for count entries
+   * @param max_min Maximum wait time in minutes
+   * @param updatedRowsCount Expected count of rows
+   */
+  public static void waitForRowsUpdate(Connection connection, String query, int max_min,
+      int updatedRowsCount) throws SQLException {
+    final Duration TIME_INTERVAL = Duration.ofSeconds(30);
+    final Duration MAX_TIME = Duration.ofMinutes(max_min);
+    for (long waitTime = 0; waitTime <= MAX_TIME.getSeconds(); waitTime += TIME_INTERVAL.getSeconds()) {
+      if (getEntriesCountForQuery(connection, query) >= updatedRowsCount) {
+        break;
+      }
+      try {
+        Thread.sleep(TIME_INTERVAL.toMillis());
+      } catch (InterruptedException e) {
+        LOGGER.error("Interruption exception");
```

Comment:
Usually you want to propagate an interruption exception by calling `Thread.currentThread().interrupt()`. Otherwise, this will potentially result in ignoring terminate signals.

Reply:
Corrected, please review